src/v/kafka/server/group.cc

```diff
@@ -1201,7 +1201,9 @@ void group::remove_pending_member(kafka::member_id member_id) {
     }
 }
 
-void group::shutdown() {
+ss::future<> group::shutdown() {
+    _auto_abort_timer.cancel();
+    co_await _gate.close();
     // cancel join timer
```

Comment:
nit: in future PRs please be careful to make sure that each commit stands on its own and is internally consistent. these two lines would seem to make more sense in the previous commit. without them, the previous commit starts a background fiber and never stops that fiber. the previous commit would compile, but has a bug that would likely lead to segfault.